src/main/groovy/com/google/appengine/AppEnginePlugin.groovy

```diff
@@ -245,28 +256,32 @@ class AppEnginePlugin implements Plugin<Project> {
         }
     }
 
-    private void configureAppConfig(Project project, AppEnginePluginExtension appEnginePluginExtension) {
+    private void configureWorkingDirectory(Project project) {
+        AppConfigTaskTemplate.metaClass.getWorkingDirectory = { -> this.explodedAppDirectory }
```

Comment:
In regard to `this`, for example, the property `explodedSdkDirectory` clashes with getter method `getExplodedSdkDirectory` and results in `StackOverflowError` in `appengineDownloadSdk` task. That was the reason to guard all properties instead of digging to the each particular case.